src/GCDWheel.cs

```diff
@@ -116,6 +140,35 @@ public unsafe class GCDWheel {
             }
         }
 
+        private void GCDTimeoutHelper(Configuration conf){
+            //create isRunning bool and use it to create an idle "timer"
+            isRunning = (DataStore.Action->ElapsedGCD != DataStore.Action->TotalGCD) || HelperMethods.IsCasting();
+            if (!isRunning && idleTimer < 25 * conf.GCDTimeout) idleTimer++;
```

Comment:
There shouldn't be a migration issue -- it will only apply to the two of us.  For all the end users it should always be an int.

It's an int here to make any comparisons we do against the timeout predictable.  Otherwise we have to do with floating point error.  Up to you in the end.

I thought about doing as you outline, but I wasn't sure of the performance implication vs. the actual benefit.  Does it really matter how long the timeout is if users are going to adjust to taste anyway?  The counter was just so simple that it appealed to me.  Not having to do a bunch of float math when we can just increment an int and achieve 90% of the results.

Reply:
Gotcha. Forgot the GCDTimeout was not released yet. No issue there.
My points for moving to ms are:
 - Timeout multiplier is a harder concept to understand.
 - People with low end setups may have irregular behavior. (Such as different timings in raids versus dummy due to computer skipping frames)
But yeah, you are right most people will probably not notice the difference so we can keep it like this for now.

Reply:
Conceptually, I think using real time makes more sense. If you don't think it's a performance hit in itself to hit the system clock every update and do some float math, I'm not bothered by that at all.

As you say, it's what people would expect.  I just wasn't sure about the performance.  Novice programmer and all.

Reply:
The performance hit shouldn't be that high. And I have been keeping track of the update time and we are still under reasonable levels for average draw time, most of the time is actually spent in imgui calls so I would say to go ahead and move it to ms if possible.

Reply:
I've got a working timer based system based on:

```
            if (!idleAccumDone &&!isRunning && idleAccumulator < (int)(1000 * conf.GCDTimeoutMul))  
                idleAccumulator += framework.UpdateDelta.Milliseconds;  
```

But I don't really like it.  Because we are counting the ms between iterations, we don't hit every value in the counter.  We skip some because most iterations are longer than 1ms.  As a result, we end up having to use 

idleAccumulator >= Timeout && idleAccumulation < Timeout - (some arbitrary number).

 instead of :

idle Accumulator == Timeout 

for all of our tests.  This is most important for the ABC check, but comes into play in a couple of other places.  We *really* need that conditional to go false before the alert animation finishes, or it bugs out.  So it was super handy having it only ever true for one iteration.

That or have an elaborate system of bools that get triggered when we pass checkpoints in the loop and reset as needed.  Which usually takes the form of being reset on button press (I've been avoiding this as much as possible, in case someone likes to mash buttons)  The GCD end function happens too early to be useful for the reset of the ABC alert.

I'm gonna sleep on this and have another go tomorrow with fresh eyes.  I'll let you know.

Reply:
Have a look at:

https://github.com/CaiClone/GCDTracker/commit/98dafe7a0b908fb258aaa0a1c982dccf35172ab3

I'm not like super thrilled with it.  Feels clunky.  Maybe it's just in my head.

Reply:
I like it a lot! The idle accumulator feels easier to understand this way for me and we are also doing the correct thing to avoid weird edge cases. 
This looks good to merge!

Reply:
Check this PR, I just pushed a more refined version of it to my main.  Let me know if this is at least as good as the one linked above -- I think I prefer it but we can do whichever you like better.